OpenNefia.Content/UI/Element/FeatWindow.cs

```diff
@@ -0,0 +1,311 @@
+﻿using Love;
+using OpenNefia.Content.Inventory;
+using OpenNefia.Content.Prototypes;
+using OpenNefia.Content.UI.Element.List;
+using OpenNefia.Core.Audio;
+using OpenNefia.Core.Input;
+using OpenNefia.Core.IoC;
+using OpenNefia.Core.Locale;
+using OpenNefia.Core.Maths;
+using OpenNefia.Core.Prototypes;
+using OpenNefia.Core.Rendering;
+using OpenNefia.Core.UI.Element;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OpenNefia.Content.UI.Element
+{
+    public class FeatWindow : UiWindow
+    {
+        public abstract record FeatData
+        {
+            public virtual string Name { get; } = "";
+            public virtual string Description { get; } = "";
+            public virtual Core.Maths.Color Color { get; } = Core.Maths.Color.Black;
+            public record Feat(FeatPrototype Prototype, int Level) : FeatData
+            {
+                public virtual int TotalLevel => Level + 1;
+                public override string Name => Loc.GetPrototypeString(Prototype.GetStrongID(), $"{Math.Min(TotalLevel, Prototype.LevelMax)}.Name")! 
+                    + (Level >= Prototype.LevelMax ? $"({Loc.GetString("Elona.FeatMenu.FeatMax")})" : string.Empty);
+                public override string Description => Loc.GetPrototypeString(Prototype.GetStrongID(), "MenuDesc")!;
+
+                public override Core.Maths.Color Color => Level switch
+                {
+                    > 0 => UiColors.CharaMakeStatLevelBest,
+                    < 0 => UiColors.CharaMakeStatLevelSlight,
+                    _ => base.Color,
+                };
+            }
+            public record FeatHeader(string LocalizeText) : FeatData
+            {
+                public override string Name => Loc.GetString(LocalizeText);
+            }
+            public record GainedFeat : Feat
+            {
+                public override int TotalLevel => Level;
+                public override string Name => $"[{Loc.GetString($"Elona.FeatMenu.FeatType.{Prototype.FeatType}")}]{Description}";
+
+                public override string Description => Loc.GetPrototypeString(Prototype.GetStrongID(), $"{Math.Min(TotalLevel, Prototype.LevelMax)}.Desc")!;
+                public bool IsRace;
```

Comment:
What about other feat categories (mutation, ether disease)?

Reply:
yeah the field was redundant, removed it

Reply:
changed in b1407d8